### PR TITLE
Updates to support MPAS applications (RRFS and HFIP)

### LINKED ISSUE #1030: MPAS applications: add processing for lat–lon projection, add soil fields (TSOIL, SOILW)
The INITPOST_MPAS interface (adapted from INITPOST) does not currently process model output on a latitude–longitude projection (i.e., MAPTYPE = 0).  This projection is used in GSL's modeling system for the Hurricane Forecast Improvement Project (HFIP–MPAS).  UPP produces GRIB2 files for this model, but the grid lengths are decoded by wgrib2 as "0".  For this reason, the GRIB2 files are not currently useable from HFIP–MPAS.

Additionally, soil temperature (TSOIL) and and soil moisture (SOILW) are missing from the GRIB2 output in both the HFIP–MPAS and RRFS–MPAS.  It appears that the netCDF "ZS" array (depths of the soil layers) is not being created or read-in correctly.

### PR DESCRIPTION
**Pull request by jaymes-kenyon**
This PR provides several updates to improve UPP functionality with MPAS (specifically MPASSIT) output, including:
1) Changes in INITPOST_MPAS to handle lat–lon projections (as used in HFIP)   
2) Inclusion of SOILT and SOILW output
3) Silencing various "sample" print statements

Testing has been successful in the HFIP-MPAS and RRFS-MPAS configurations, both of which use MPASSIT-generated netCDF output.  This PR addresses issue #1030.

### CODE REVIEW

**Comment by WenMeng-NOAA** [comment]
@jaymes-kenyon Can you open a UPP issue to link to this PR?

File: sorc/ncep_post.fd/WRFPOST.F [hunk 1]
```diff
@@ -384,6 +384,12 @@ PROGRAM WRFPOST
 ! Read and set global value for surface physics scheme
           call ext_ncd_get_dom_ti_integer(DataHandle                      &
             ,'SF_SURFACE_PHYSICS',itmp,1,ioutcount, status )
+
+          ! J. Kenyon / 23 Aug 2024: GSL MPAS output erroneously
+          ! indicates SF_SURFACE_PHYSICS = 0 (should be 3), so overwrite
+          ! here:
+          if (modelname == 'RAPR' .and. submodelname == 'MPAS') itmp = 3 
+
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/WRFPOST.F:393]: @jaymes-kenyon Please add a change log the beginning session of this file (line 40).

File: sorc/ncep_post.fd/INITPOST_MPAS.F [hunk 2]
```diff
@@ -2,7 +2,7 @@
 !> @brief initpost() initializes post for run.
 !>
 !> @author Russ Treadon W/NP2 @date 1993-11-10
-
+!
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/INITPOST_MPAS.F:5]: @jaymes-kenyon Please add a change log. You should be listed as the author who initialized this subroutine.

**Comment by WenMeng-NOAA** [comment]
@jaymes-kenyon Do you have MPAS-HFIP model output for my UPP standalone test? Would MPAS output you provided from last PR be sufficient for my test for RRFS-MPAS?

**Comment by jaymes-kenyon** [comment]
Hi @WenMeng-NOAA — I've placed some HFIP-MPAS output on Hera, along with an "itag" file:
`/scratch2/BMC/wrfruc/kenyon/sample/hfip_mpas/2024080812`
The intended UPP control file for HFIP-MPAS is `postxconfig-NT-hfip.txt`

For RRFS-MPAS testing, you may continue to use the sample from my previous PR.  Thanks!

**Comment by WenMeng-NOAA** [comment]
@jaymes-kenyon Please validate my UPP standalone tests on Hera:
```
1) MPAS data: /home/Wen.Meng/stmp2/mpas_2024082112
2) MPAS_HFIP data: /home/Wen.Meng/stmp2/mpas_2024080900
```

**Comment by jaymes-kenyon** [comment]
Hi @WenMeng-NOAA — Your tests look good to me—thanks for sharing!

**Comment by WenMeng-NOAA** [comment]
@jaymes-kenyon There seem to be an excessive number of printouts from INITPOST_MPAS.F. Could you reduce them?
e.g.
```
         if(i==im/2.and.j==(jsta+jend)/2)                           &
        print*,'i,j,L,ZINT from unipost= ',i,j,LM+1,ZINT(I,J,LM+1)      &
              , ALPINT(I,J,LM+1),ALPINT(I,J,LM)
```

**Comment by jaymes-kenyon** [comment]
@WenMeng-NOAA — Thanks!  I commented-out several print statements in the most recent commit.  In my test, this reduced the size of the UPP log file by about 20%.  We can probably find many more "print" and "write" statements to comment-out going forward.

**Comment by WenMeng-NOAA** [comment]
@jaymes-kenyon For [this print statement](https://github.com/jaymes-kenyon/UPP/blob/mpas/sorc/ncep_post.fd/INITPOST_MPAS.F#L128), I would suggest printing information exclusively from process 0 as:
```
if (me==0) WRITE(6,*)'INITPOST_MPAS:  ENTER INITPOST_MPAS'
```

**Comment by WenMeng-NOAA** [comment]
@jaymes-kenyon I would suggest this [printout](https://github.com/jaymes-kenyon/UPP/blob/mpas/sorc/ncep_post.fd/INITPOST_MPAS.F#L2135) come exclusively from process 0.

**Comment by jaymes-kenyon** [comment]
@WenMeng-NOAA — Thanks for the advice; I added this logic to several "write" statements.

**Comment by WenMeng-NOAA** [comment]
@jaymes-kenyon I would suggest this type of printout (print*,***)in INITPOST_MPAS.F come exclusively from process 0.

**Comment by WenMeng-NOAA** [comment]
@jaymes-kenyon I would suggest the printout (write(*,*)) in getVariable.f come exclusively from process 0.
You may add process id "me" in this [line](https://github.com/jaymes-kenyon/UPP/blob/mpas/sorc/ncep_post.fd/getVariable.f#L54).

**Comment by jaymes-kenyon** [comment]
Thanks, @WenMeng-NOAA — I will make another commit soon with these changes to INITPOST_MPAS and getVariable.  Should I also modify the write statements in **getIVariableN.f**?  It is very similar to getVariable.f

**Comment by WenMeng-NOAA** [comment]
> Thanks, @WenMeng-NOAA — I will make another commit soon with these changes to INITPOST_MPAS and getVariable. Should I also modify the write statements in **getIVariableN.f**? It is very similar to getVariable.f

@jaymes-kenyon It would be great if you apply the changes in getIVariableN.f. Thanks!

**Comment by WenMeng-NOAA** [comment]
@jaymes-kenyon The printouts in the runtime log have been reduced by more than 50%. Thanks for working on this. We will proceed with your PR for final testing.

**Comment by WenMeng-NOAA** [comment]
@FernandoAndrade-NOAA Please start the UPP RTs on R&D machines. There should be no baseline update.

**Comment by WenMeng-NOAA** [comment]
The UPP RTs have been completed on WCOSS2. No baseline update is needed.

**Review by FernandoAndrade-NOAA — APPROVED** [review]
No changes in RT results across Hera, Hercules, and Orion.

**Review by WenMeng-NOAA — APPROVED** [review]

**Comment by WenMeng-NOAA** [comment]
This PR is ready for merging.

**Pull request merged**